governance/program/src/instruction.rs

```diff
@@ -140,58 +130,64 @@ pub enum GovernanceInstruction {
         transfer_upgrade_authority: bool,
     },
 
-    /// Create Proposal account for Instructions that will be executed at various slots in the future
-    /// The instruction also grants Admin and Signatory token to the provided account
+    /// Creates Proposal account for Instructions that will be executed at various slots in the future
     ///
-    ///   0. `[writable]` Uninitialized Proposal account
-    ///   1. `[writable]` Initialized Governance account
-    ///   2. `[writable]` Initialized Signatory Mint account
-    ///   3. `[writable]` Initialized Admin Mint account
-    ///   4. `[writable]` Initialized Admin account for the issued admin token
-    ///   5. `[writable]` Initialized Signatory account for the issued signatory token
-    ///   6. '[]` Token program account
-    ///   7. `[]` Rent sysvar
+    ///   0. `[writable]` Proposal account. PDA seeds ['governance',account_governance, governing_token_mint, name]
+    ///   1. `[writable]` Governance
+    ///   2. `[]` Token Owner Record account
+    ///   3. `[signer]` Governance Authority (Token Owner or Governance Delegate)    
+    ///   4. `[signer]` Payer
+    ///   5. `[]` System program
+    ///   6. `[]` Rent sysvar
+    ///   7. `[]` Clock sysvar    
     CreateProposal {
-        #[allow(dead_code)]
-        /// Link to gist explaining proposal
-        description_link: String,
-
         #[allow(dead_code)]
         /// UTF-8 encoded name of the proposal
         name: String,
 
         #[allow(dead_code)]
-        /// The Governing token (Community or Council) which will be used for voting on the Proposal
-        governing_token_type: GoverningTokenType,
+        /// Link to gist explaining proposal
+        description_link: String,
+
+        #[allow(dead_code)]
+        /// Governing Token Mint the Proposal is created for
+        governing_token_mint: Pubkey,
     },
 
-    /// [Requires Admin token]
     /// Adds a signatory to the Proposal which means this Proposal can't leave Draft state until yet another Signatory signs
-    /// As a result of this call the new Signatory will receive a Signatory Token which then can be used to Sign proposal
     ///
     ///   0. `[writable]` Proposal account
-    ///   1. `[writable]` Initialized Signatory account
-    ///   2. `[writable]` Initialized Signatory Mint account
-    ///   3. `[signer]` Admin account
-    ///   4. '[]` Token program account
-    AddSignatory,
+    ///   1. `[]` Token Owner Record account
+    ///   2. `[signer]` Governance Authority (Token Owner or Governance Delegate)    
+    ///   3. `[writable]` Signatory Record Account
+    ///   4. `[signer]` Payer
+    ///   5. `[]` System program
+    ///   6. `[]` Rent sysvar
+    AddSignatory {
+        #[allow(dead_code)]
+        /// Signatory to add to the Proposal
+        signatory: Pubkey,
```

Comment:
I kind of was thinking the other way was anti-pattern :). Since you can pass none existing account then using it to get just the key feels wrong to me. 
I'm not sure what the overhead on the runtime is for passing accounts, but I guess it's negligible. However I find it confusing to pass accounts only to us them for keys.
However if you think it's better to make it a rule and always pass accounts as accounts, even to just get the key, then that's fine with me an would eliminate some decisions to make. Please let me know.

Reply:
I don't think there's any consensus about what's better, to be honest.  Evey added account is extra overhead for the runtime since it needs to fetch it.  We've had issues of people inputting bad pubkeys in the past for irreversible operations, so we were recommended to pass in accounts as much as possible.  In this case, it's totally reversible, so it should be fine to leave it if you prefer it.

Reply:
Ok, thank you for explanation. I'll leave it for now but I'll keep watching that use case